alkira/resource_alkira_virtual_network_manager_azure.go

```diff
@@ -0,0 +1,207 @@
+package alkira
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAlkiraVirtualNetworkManagerAzure() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Manager Virtual Network Manager for Azure.",
+		CreateContext: resourceVirtualNetworkManagerAzureCreate,
+		UpdateContext: resourceVirtualNetworkManagerAzureUpdate,
+		DeleteContext: resourceVirtualNetworkManagerAzureDelete,
+		ReadContext:   resourceVirtualNetworkManagerAzureRead,
+		CustomizeDiff: func(ctx context.Context, d *schema.ResourceDiff, m interface{}) error {
+			client := m.(*alkira.AlkiraClient)
+
+			old, _ := d.GetChange("provision_state")
+
+			if client.Provision == true && old == "FAILED" {
+				d.SetNew("provision_state", "SUCCESS")
+			}
+
+			return nil
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the Alkira Azure Virtual Network Manager.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"region": {
+				Description: "The Azure Region where the Azure Virtual Network" +
+					" Manager will be created. eg: `eastus`.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"subscription_id": {
+				Description: "The ID of the Azure Subscription account in which the" +
+					" Azure Virtual Network Manager is created.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cxp": {
+				Description: "The CXP on which the virtual network manager is created.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"resource_group": {
+				Description: "The name of the Azure Resource Group in which the" +
+					" Azure Virtual Network Manager is created.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Description: "The description of the Alkira Azure Virtual Network Manager.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"credential_id": {
+				Description: "ID of credential managed by Credential Manager.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"subscriptions_in_scope": {
```

Comment:
This is a bit confusing too... We asked for subscription_id, what subscriptions in scope means...

Reply:
The subscription id is for the subscription that the AVNM will be created in and the `subscriptions_in_scope` are the subscriptions managed by the AVNM.

Reply:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/network_manager